### ISSUE TITLE
unregisterReceiver()报错

### ISSUE THREAD

**Issue by Tamarackwyh** [osmdroid/osmdroid#1916]
## Issue Type
bug

## Description

关闭软件后弹出IntentReceiverLeaked错误信息：
android.app.IntentReceiverLeaked: Activity com.deepiot.deepiotpatrol.LocationMap.PatorlLocationMapLineActivity has leaked IntentReceiver org.osmdroid.tileprovider.modules.MapTileFileStorageProviderBase$MyBroadcastReceiver@aa16a01 that was originally registered here. Are you missing a call to unregisterReceiver()?

报错提示在这一行上:
MapTileProviderBase gaodeRoadNetProvider = new MapTileProviderBasic(getContext(), MapConstant.GaodeRoadNet);

## code to reproduce the problem

        IRegisterReceiver registerReceiver = new SimpleRegisterReceiver(getContext());
        TileWriter tileWriter = new TileWriter();
        MapTileFilesystemProvider fileSystemProvider = new MapTileFilesystemProvider(registerReceiver, TileSourceFactory.getTileSource(mapType));
        MapTileFileArchiveProvider fileArchiveProvider = new MapTileFileArchiveProvider(registerReceiver, TileSourceFactory.getTileSource(mapType), findArchiveFiles(mapType));
        MapTileDownloader downloaderProvider = new MapTileDownloader(TileSourceFactory.getTileSource(mapType), tileWriter, new NetworkAvailabliltyCheck(getContext()));
        MapTileProviderArray tileProviderArray = new MapTileProviderArray(TileSourceFactory.getTileSource(mapType), registerReceiver, new MapTileModuleProviderBase[]{fileSystemProvider, fileArchiveProvider, downloaderProvider});
        getTileProvider().clearTileCache();
        setTileSource(TileSourceFactory.getTileSource(mapType));
        setTileProvider(tileProviderArray);
        if (this.TianDiTuBiaoZhuLayer != null) {
            tianDiTuMapBiaoZhuOverlay.remove(this.TianDiTuBiaoZhuLayer);
            this.TianDiTuBiaoZhuLayer = null;
        }
        MapTileProviderBase gaodeRoadNetProvider = new MapTileProviderBasic(getContext(), MapConstant.GaodeRoadNet);
        if (gaodeRoadNetProvider != null) {
            this.TianDiTuBiaoZhuLayer = new TilesOverlay(gaodeRoadNetProvider, getContext());
            this.TianDiTuBiaoZhuLayer.setLoadingBackgroundColor(0);
            this.TianDiTuBiaoZhuLayer.setLoadingLineColor(0);
            //添加到地图图层
            tianDiTuMapBiaoZhuOverlay.add(this.TianDiTuBiaoZhuLayer);
        }

### If it's a bug, version(s) of android this affects:
 minSdkVersion 19
targetSdkVersion 30

### Version of osmdroid the issue relates to:
implementation 'org.osmdroid:osmdroid-android:6.1.11@aar'

**Comment by eclectice** [comment]
https://github.com/osmdroid/osmdroid/issues/1666
https://github.com/osmdroid/osmdroid/issues/562
https://github.com/osmdroid/osmdroid/issues/343

**Comment by Tamarackwyh** [comment]
Method invalid

**Comment by Tamarackwyh** [comment]
> Check the proper way to detach the map which will eventually call the `unregisterReceiver()` method.
> 
> #1666 #562 #343
> 
> On StackOverflow: https://stackoverflow.com/questions/37304905/leaked-intentreceiver-exception-using-osmdroid

Method invalid

**Comment by Tamarackwyh** [comment]
![3FA771FF-B68D-4f68-AACA-757EF64D50B4](https://github.com/osmdroid/osmdroid/assets/45580830/7f8eee1c-edf8-4474-9cac-c68b8cfaed8a)
I have never used registerReceiver, why there is a prompt unregisterReceiver？

**Comment by Tamarackwyh** [comment]
[issue was settled]

if (gaodeRoadNetProvider != null) {
    gaodeRoadNetProvider.detach();
}
